### DISCUSSION TITLE
Autogenerated migration script does not include postgreSQL schema name in enum type

### DISCUSSION THREAD

**Discussion by TheShubhendra** [sqlalchemy/alembic#1417] [Usage Questions] [answered]
I have autogenerated a migration script for postgresSQL . It includes schame name in new tables but not in new Enum.
```python3
def upgrade() -> None:
    # ### commands auto generated by Alembic - please adjust! ###
    op.create_table('accounts',
    sa.Column('id', sa.Integer(), nullable=False),
    sa.Column('status', sa.Enum('active', 'inactive', name='accountstatus'), nullable=True),
    sa.PrimaryKeyConstraint('id'),
    schema='auth'
    )
```
 ```python3
 cursor.execute(statement, parameters)
sqlalchemy.exc.ProgrammingError: (psycopg2.errors.InvalidSchemaName) no schema has been selected to create in

[SQL: CREATE TYPE accountstatus AS ENUM ('active', 'inactive')]
(Background on this error at: https://sqlalche.me/e/20/f405)
```

If i change line `sa.Column('status', sa.Enum('active', 'inactive', name='accountstatus'), nullable=True),` to `   sa.Column('status', sa.Enum('active', 'inactive', name='accountstatus', schema='auth'), nullable=True),` It works!

**Comment by CaselIT** [comment, marked as answer]
Hi,

You need to set https://docs.sqlalchemy.org/en/20/core/type_basics.html#sqlalchemy.types.Enum.params.inherit_schema to true fir the enum to inherit the schema.

@zzzeek why is that flag not set to true by default? Could it be a change for sqlalchemy 2.1?

**Reply by TheShubhendra** [reply]
Ok Thanks a lot!

**Comment by zzzeek** [comment]
> Hi,
> 
> You need to set https://docs.sqlalchemy.org/en/20/core/type_basics.html#sqlalchemy.types.Enum.params.inherit_schema to true fir the enum to inherit the schema.
> 
> @zzzeek why is that flag not set to true by default? Could it be a change for sqlalchemy 2.1?

yes let's do this

**Reply by CaselIT** [reply]
ah look at that, it was already there! https://github.com/sqlalchemy/sqlalchemy/issues/10594
